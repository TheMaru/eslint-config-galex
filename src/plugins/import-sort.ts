import { type RulesetCreator } from '../types';

export const createSimpleImportSortRules: RulesetCreator = ({
  rules: customRules,
}) => {
  return {
    'simple-import-sort/imports': 'warn',
    'simple-import-sort/exports': 'warn',
    ...customRules,
  };
};
