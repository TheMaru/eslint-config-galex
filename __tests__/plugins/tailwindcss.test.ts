import { createTailwindPlugin } from '../../src/plugins/tailwindcss';
import { defaultProject } from '../shared';

test('allows passing rules', () => {
  const ruleName = 'foo';
  const ruleValue = 'off';

  const result = createTailwindPlugin({
    ...defaultProject,
    rules: {
      [ruleName]: ruleValue,
    },
  });

  expect(result?.[ruleName]).toBe(ruleValue);
});

test('with tailwind', () => {
  expect(
    createTailwindPlugin({
      ...defaultProject,
      hasTailwind: true,
    })
  ).toMatchSnapshot();
});
