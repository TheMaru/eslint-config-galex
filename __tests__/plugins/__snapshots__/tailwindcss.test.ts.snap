// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with tailwind 1`] = `
{
  "tailwindcss/classnames-order": "warn",
  "tailwindcss/enforces-negative-arbitrary-values": "warn",
  "tailwindcss/enforces-shorthand": "warn",
  "tailwindcss/migration-from-tailwind-2": "warn",
  "tailwindcss/no-arbitrary-value": "off",
  "tailwindcss/no-contradicting-classname": "error",
  "tailwindcss/no-custom-classname": "warn",
}
`;
